tests/testCoaddInputs.py

```diff
@@ -0,0 +1,256 @@
+
+
+class MockCoadd(object):
+    """Class to make a mock coadd
+    """
+    def __init__(self, numExp, version=2):
+        """Create a coadd with the specified number of exposures.
+
+        @param[in] numExp  total number of exposures that will be made
+        @param[in] version  desired version of CoaddInput/ExposureTable; see MockExposure for details
+
+        Useful fields include:
+        - coadd
+        - exposures  a list of exposures that went into the coadd
+        """
+        coaddInputRecorder = CoaddInputRecorderTask(name="coaddInputRecorder")
+        mockExposure = MockExposure(numExp=numExp, coaddInputRecorder=coaddInputRecorder, version=version)
+        self.exposures = [mockExposure.makeExposure(i) for i in range(numExp)]
+
+        exp0 = self.exposures[0]
+        self.coadd = lsst.afw.image.ExposureF(exp0.getBBox())
+        self.coadd.setWcs(exp0.getWcs())
+
+        coaddInputs = coaddInputRecorder.makeCoaddInputs()
+        for exp in self.exposures:
+            coaddInputRecorder.addVisitToCoadd(coaddInputs=coaddInputs, coaddTempExp=exp, weight=1.0/numExp)
+        self.coadd.getInfo().setCoaddInputs(coaddInputs)
+
+
+class CoaddInputsTestCase(lsst.utils.tests.TestCase):
+    def setUp(self):
+        self.version = 2  # desired version of ExposureTable/CoaddInput
+        self.numExp = 3
+        mockCoadd = MockCoadd(numExp=self.numExp, version=self.version)
+        self.coadd = mockCoadd.coadd
+        self.exposures = mockCoadd.exposures
+        self.dataDir = os.path.join(os.path.dirname(__file__), "data")
+
+    def tearDown(self):
+        del self.coadd
+        del self.exposures
+
+    def assertPsfsNearlyEqual(self, psf1, psf2):
+        im1 = psf1.computeImage()
+        im2 = psf2.computeImage()
+        self.assertImagesNearlyEqual(im1, im2)
+
+    def getCoaddPath(self, version):
+        return os.path.join(self.dataDir,
+                            "testCoaddInputs_coadd_with_version_{}_data.fits".format(version))
+
+    def testPersistence(self):
+        """Read and write a coadd and check the CoaddInputs"""
+        coaddPath = self.getCoaddPath(version=self.version)
+        self.coadd.writeFits(coaddPath)
+        coadd = lsst.afw.image.ExposureF(coaddPath)
+        coaddInputs = coadd.getInfo().getCoaddInputs()
+        self.assertCoaddInputsOk(coaddInputs, version=self.version)
+        if not SaveCoadd:
+            os.unlink(coaddPath)
+        else:
+            print("SaveCoadd true; saved coadd as: %r" % (coaddPath,))
+
+    def xtestReadV1Coadd(self):
+        """Read a coadd that contains version 1 CoaddInputs"""
+        coaddPath = self.getCoaddPath(version=1)
+        coadd = lsst.afw.image.ExposureF(coaddPath)
+        coaddInputs = coadd.getInfo().getCoaddInputs()
+        self.assertCoaddInputsOk(coaddInputs, version=1)
+
+    def assertCoaddInputsOk(self, coaddInputs, version):
+        self.assertIsNotNone(coaddInputs)
+        visitTable = coaddInputs.visits
```

Comment:
If you're going to test just one of these, it should be `ccds`, not `visits`; most of these quantities in the `visits` table come from the `coaddTempExp`, which is *probably* better than setting them to None but might still be confusing in some cases (e.g. "the visit WCS is the same as the coadd WCS").  The `ccds` table is what corresponds to what is in a `calexp`, and that's what most people will want to look at and hence what we want to make sure doesn't regress.